server/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java

```diff
@@ -267,18 +271,29 @@ private ImmutableList<Block> getBlocks(
   }
 
   private boolean showBlock(Block block) {
+    if (block.getRepeatedEntity().isPresent()) {
```

Comment:
suggestion (non-blocking): consider adding a commenting mentioning that this is examining of the visibility of the entity's parents